data_model/src/predicate.rs

```diff
@@ -73,20 +73,23 @@ macro_rules! nontrivial {
 }
 
 /// Predicate combinator enum.
-#[derive(Debug, Clone, Serialize, Deserialize, Encode, Decode, IntoSchema)]
+#[derive(
+    Debug, Clone, PartiallyTaggedSerialize, PartiallyTaggedDeserialize, Encode, Decode, IntoSchema,
+)]
 // Ideally we would enforce `P: PredicateTrait<Input>` here, but I
 // couldn't find a way to do it without polluting everything
 // downstream with explicit lifetimes, since we would need to
 // store PhantomData<Input> here, and `Input`s are most often
 // references (e.g. &Value).
 pub enum GenericPredicateBox<P> {
     /// Logically `&&` the results of applying the predicates.
-    And(NonTrivial<Self>),
-    /// Logically `||` the results of applying the predicates.
-    Or(NonTrivial<Self>),
+    And(NonTrivial<GenericPredicateBox<P>>),
+    /// Logically `||` the results of applying the predicats.
+    Or(NonTrivial<GenericPredicateBox<P>>),
     /// Negate the result of applying the predicate.
-    Not(Box<Self>),
+    Not(Box<GenericPredicateBox<P>>),
     /// The raw predicate that must be applied.
+    #[serde_partially_tagged(untagged)]
```

Comment:
Please revert.

Reply:
Unfortunately, currently it's limitation of partially tagged (de)serialize macros.
Reverting would require to complicate this macro with resolving `Self` which i would prefer to do in another PR if possible.